XamarinCommunityToolkit/Views/DrawingView/DrawingView.shared.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public class DrawingView : View
+	{
+		const int minValueGranularity = 5;
+
+		public static readonly BindableProperty GranularityProperty =
+			BindableProperty.Create(nameof(Granularity), typeof(int), typeof(DrawingView), minValueGranularity);
+
+		public static readonly BindableProperty EnableSmoothedPathProperty =
+			BindableProperty.Create(nameof(EnableSmoothedPath), typeof(bool), typeof(DrawingView), default(bool));
+
+		public static readonly BindableProperty LineColorProperty =
+			BindableProperty.Create(nameof(LineColor), typeof(Color), typeof(DrawingView), Color.Default);
+
+		public static readonly BindableProperty LineWidthProperty =
+			BindableProperty.Create(nameof(LineWidth), typeof(float), typeof(DrawingView), 5f);
+
+		public static readonly BindableProperty ClearOnFinishProperty =
+			BindableProperty.Create(nameof(ClearOnFinish), typeof(bool), typeof(DrawingView), default(bool));
+
+		public static readonly BindableProperty PointsProperty = BindableProperty.Create(
+			nameof(Points), typeof(IEnumerable<Point>), typeof(DrawingView), default(IEnumerable<Point>),
+			BindingMode.TwoWay);
+
+		public static readonly BindableProperty DrawingCompletedCommandProperty = BindableProperty.Create(
+			nameof(DrawingCompletedCommand), typeof(ICommand), typeof(DrawingView), default(ICommand));
+
+		public IEnumerable<Point> Points
+		{
+			get => (IEnumerable<Point>)GetValue(PointsProperty);
+			set
+			{
+				SetValue(PointsProperty, value);
+				if (value.Any())
+				{
+					if (DrawingCompletedCommand != null && DrawingCompletedCommand.CanExecute(null))
+					{
+						DrawingCompletedCommand.Execute(value);
+					}
+				}
+
+				if (ClearOnFinish)
+				{
+					SetValue(PointsProperty, Enumerable.Empty<Point>());
+				}
+			}
+		}
+
+		public ICommand DrawingCompletedCommand
+		{
+			get => (ICommand)GetValue(DrawingCompletedCommandProperty);
+			set => SetValue(DrawingCompletedCommandProperty, value);
+		}
+
+		public int Granularity
+		{
+			get => (int)GetValue(GranularityProperty);
+			set
+			{
+				if (value >= minValueGranularity)
```

Comment:
Typically such checks should be done in static bindable property

See example: https://github.com/xamarin/XamarinCommunityToolkit/blob/97bed3b08ea13391144cc45536e86ced9f386cae/XamarinCommunityToolkit/Views/RangeSlider.shared.cs#L46

coerceValue delegate.